yubikit/support.py

```diff
@@ -0,0 +1,434 @@
+# Copyright (c) 2015-2022 Yubico AB
+# All rights reserved.
+#
+#   Redistribution and use in source and binary forms, with or
+#   without modification, are permitted provided that the following
+#   conditions are met:
+#
+#    1. Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#    2. Redistributions in binary form must reproduce the above
+#       copyright notice, this list of conditions and the following
+#       disclaimer in the documentation and/or other materials provided
+#       with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+from .core import (
+    TRANSPORT,
+    YUBIKEY,
+    PID,
+    Version,
+    Connection,
+    NotSupportedError,
+    ApplicationNotAvailableError,
+)
+from .core.otp import OtpConnection, CommandRejectedError
+from .core.fido import FidoConnection
+from .core.smartcard import (
+    AID,
+    SmartCardConnection,
+    SmartCardProtocol,
+)
+from .management import (
+    ManagementSession,
+    DeviceInfo,
+    DeviceConfig,
+    USB_INTERFACE,
+    CAPABILITY,
+    FORM_FACTOR,
+    DEVICE_FLAG,
+)
+from .yubiotp import YubiOtpSession
+
+from time import sleep
+from typing import Optional, Tuple
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def _otp_read_data(conn) -> Tuple[Version, Optional[int]]:
+    otp = YubiOtpSession(conn)
+    version = otp.version
+    serial: Optional[int] = None
+    try:
+        serial = otp.get_serial()
+    except Exception:
+        logger.debug("Unable to read serial over OTP, no serial", exc_info=True)
+    return version, serial
+
+
+AID_U2F_YUBICO = b"\xa0\x00\x00\x05\x27\x10\x02"  # Old U2F AID
```

Comment:
In other place `bytes.fromhex("a0000005272001")` is used. It would be nice to keep the same format everywhere.